tests/rust.rs

```diff
@@ -0,0 +1,85 @@
+use anyhow::Context;
+
+use std::path::PathBuf;
+
+struct RustTestCase {
+    name: String,
+    humble_spec: PathBuf,
+    humble_rust_out: PathBuf,
+    main: PathBuf,
+}
+
+impl RustTestCase {
+    fn run(&self) {
+        let spec_file = std::fs::read_to_string(&self.humble_spec).expect("read humble spec");
+        let spec = humblegen::parse_spec_str_or_panic(&spec_file);
+        let spec_rust = humblegen::Language::Rust.render(&spec).to_string();
+        // TODO rustfmt
+        std::fs::write(&self.humble_rust_out, spec_rust).expect("write generated rust code");
+
+        let t = trybuild::TestCases::new();
+        t.pass(&self.main);
+        // cases run on drop of t
+        drop(t);
+    }
+
+    fn from_test_dir(dir: &std::fs::DirEntry) -> anyhow::Result<RustTestCase> {
+        let entries: Vec<std::fs::DirEntry> = std::fs::read_dir(dir.path())?
+            .collect::<std::io::Result<Vec<std::fs::DirEntry>>>()
+            .unwrap()
```

Comment:
I'd `expect` here at the very least, but this function already returns an `anyhow` result, why not `.context` it and `?`?